crates/topos-tce/src/lib.rs

```diff
@@ -135,18 +135,18 @@ pub async fn run(
 
     let (network_client, event_stream, unbootstrapped_runtime) = topos_p2p::network::builder()
         .peer_key(key)
-        .listen_addresses(config.listen_addresses.clone())
+        .listen_addresses(config.p2p.listen_addresses.clone())
         .minimum_cluster_size(config.minimum_cluster_size)
-        .public_addresses(config.public_addresses.clone())
+        .public_addresses(config.p2p.public_addresses.clone())
         .known_peers(&boot_peers)
         .grpc_context(grpc_context)
-        .is_bootnode(config.is_bootnode)
+        .is_bootnode(config.p2p.is_bootnode)
         .build()
         .await?;
 
     debug!("Starting the p2p network");
     let network_runtime = tokio::time::timeout(
-        config.network_bootstrap_timeout,
+        Duration::from_secs(config.network_bootstrap_timeout),
```

Comment:
Why is it better to timeout here as opposed to retry periodically until it can connect?

Reply:
It'll maybe change soon with the work on the benchmarks, but the goal of this PR is not to refactor how the network behave